opennlp-tools/src/main/java/opennlp/tools/namefind/RegexNameFinderFactory.java

```diff
@@ -1,199 +1,170 @@
-/*
- * Copyright 2014 The Apache Software Foundation.
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *      http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package opennlp.tools.namefind;
-
-import java.util.HashMap;
-import java.util.Map;
-import java.util.regex.Pattern;
-
-import opennlp.tools.util.Span;
-
-/**
- *
- * Returns a RegexNameFinder based on A selection of
- * defaults or a configuration and a selection of defaults
- */
-public class RegexNameFinderFactory {
-
-  /**
-   * Allows for use of selected Defaults as well as regexes from external
-   * configuration
-   *
-   * @param config   a map where the key is a type, and the value is a
-   *                 Pattern[]. If the keys clash with default keys, the config
-   *                 map will win
-   * @param defaults the OpenNLP default regexes
-   * @return
-   */
-  public static synchronized RegexNameFinder getDefaultRegexNameFinders(Map<String, Pattern[]> config, DEFAULT_REGEX_NAME_FINDER... defaults) {
-    if (config == null) {
-      throw new IllegalArgumentException("config Map cannot be null");
-    }
-    Map<String, Pattern[]> defaultsToMap = new HashMap<>();
-    if (defaults != null) {
-      defaultsToMap = defaultsToMap(defaults);
-    }
-    defaultsToMap.putAll(config);
-    return new RegexNameFinder(defaultsToMap);
-  }
-
-  /**
-   * Returns a RegexNamefinder that will utilize specified default regexes.
-   *
-   * @param defaults the OpenNLP default regexes
-   * @return
-   */
-  public static synchronized RegexNameFinder getDefaultRegexNameFinders(DEFAULT_REGEX_NAME_FINDER... defaults) {
-    if (defaults == null) {
-      throw new IllegalArgumentException("defaults cannot be null");
-    }
-    return new RegexNameFinder(defaultsToMap(defaults));
-  }
-
-  private synchronized static Map<String, Pattern[]> defaultsToMap(DEFAULT_REGEX_NAME_FINDER... defaults) {
-    Map<String, Pattern[]> regexMap = new HashMap<>();
-    for (DEFAULT_REGEX_NAME_FINDER def : defaults) {
-      regexMap.putAll(def.getRegexMap());
-    }
-    return regexMap;
-  }
-
-  public static void main(String[] args) {
-    String text = "my email is opennlp@gmail.com and my phone num is 123-234-5678 and i like https://www.google.com and I visited MGRS  11sku528111 AKA  11S KU 528 111 and DMS 45N 123W AKA  +45.1234, -123.12 AKA  45.1234N 123.12W AKA 45 30 N 50 30 W";

```

Comment:
Someone has to write it, and spend a few minutes on it. I suggest we open a jira to do that. Right now we have many people who are motivated to pick up tasks.